core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/CaseExpression.kt

```diff
@@ -39,9 +40,9 @@ class CaseExpression<T : ValidType, U : ValidType>(
                 conditionDopeQueries.joinToString(separator = " ", prefix = " ", postfix = " ") {
                     "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}"
                 } + END,
-            parameters = caseDopeQuery.parameters + conditionDopeQueries.fold(emptyMap()) { parameters, query ->
-                parameters + query.first.parameters + query.second.parameters
-            },
+            parameters = caseDopeQuery.parameters.merge(
+                *conditionDopeQueries.map { it.first.parameters; it.second.parameters }.toTypedArray(),
```

Comment:
this is not correct, the params of first will get ignored: 
```
conditionDopeQueries.map {
    it.first.parameters; //This is just a statement and does nothing
    it.second.parameters //This is the return value of the map-function
}
```

Reply:
done, added correct merging and a test